{{cpp/title|runtime_error}}
{{cpp/error/navbar}}
{{ddcl|header=stdexcept|
class runtime_error;
}}

Defines a type of object to be thrown as exception. It reports errors that are due to events beyond the scope of the program and cannot be easily predicted. 

{{inheritance diagram/std-runtime_error}}

Exceptions of type {{tt|std::runtime_error}} are thrown by the following standard library components:
* {{lc|std::locale::locale}}
* {{lc|std::locale::combine}}
{{rrev|since=c++20|
* {{c/core|std::chrono::current_zone}}
* {{c/core|std::chrono::get_tzdb_list}}
* {{c/core|std::chrono::get_tzdb}}
* {{c/core|std::chrono::locate_zone}}
* {{c/core|std::chrono::reload_tzdb}}
* {{c/core|std::chrono::remote_version}}
* {{c/core|std::chrono::tzdb::locate_zone}}
}}

In addition, the following standard exception types are derived from {{tt|std::runtime_error}}:
* {{lc|std::range_error}}
* {{lc|std::overflow_error}}
* {{lc|std::underflow_error}}
{{rrev|since=c++11|
* {{lc|std::ios_base::failure}}
* {{lc|std::regex_error}}
* {{lc|std::system_error}}
}}
{{rrev|since=c++17|
* {{lc|std::filesystem::filesystem_error}}
}}
{{rrev|since=c++20|
* {{c/core|std::chrono::ambiguous_local_time}}
* {{c/core|std::chrono::nonexistent_local_time}}
* {{c/core|std::format_error}}
}}

===Member functions===
{{cpp/error/exception/member of derived with message|runtime_error}}

{{cpp/error/exception/inherit}}

===Defect reports===
{{dr list begin}}
{{dr list item|wg=lwg|dr=254|std=C++98|before=the constructor accepting {{c|const char*}} was missing|after=added}}
{{dr list item|wg=lwg|dr=471|std=C++98|before=the explanatory strings of {{tt|std::runtime_error}}'s&lt;br&gt;copies were implementation-defined|after=they are the same as that of the&lt;br&gt;original {{tt|std::runtime_error}} object}}
{{dr list end}}

{{langlinks|de|es|fr|it|ja|pt|ru|zh}}